extensions/schema-compare/src/schemaCompareMainWindow.ts

```diff
@@ -256,8 +258,12 @@ export class SchemaCompareMainWindow {
 			},
 		]);
 
-		// reset buttons to before comparison state
-		this.resetButtons(ResetButtonState.beforeCompareStart);
+		if (!this.sourceName || !this.targetName || this.sourceName === ' ' || this.targetName === ' ') {
```

Comment:
Is a single space actually a valid value we get here? What about other blank values?

Reply:
In an ideal scenario, there won't be blank values. And single space is indeed a valid value because these names are retrieved from "getEndpointName" function which either returns a single spaced output or a proper serverName.databaseName combo or packageFilePath, in case of dacpacs (which wouldn't be a blank value- its validity is tested way before this piece of code is triggered).

Reply:
Why not have getEndpointName return just a blank string? Or even undefined? Is there a reason you want to use a space over empty string/undefined?

Reply:
The current code had a space already, so I just tagged along with that. But a blank string makes much more sense. And that would eventually cut down the comparisons by half here. I will fix that.

Reply:
My bad, I tried doing that yesterday as well (and today again). The information "populator" for Source and Target can't handle blank values well. It just pushes out some garbage object if the value is null, but works well for a single space. Keeping it as is for now.

![image](https://user-images.githubusercontent.com/57200045/70283140-07575d80-1775-11ea-8479-8566dbf353c1.png)